src/lib/plugins/tracery/custom-modifiers.js

```diff
@@ -12,19 +13,95 @@ function noSpaces(s) {
   return s.replace(/\s+/g, '');
 }
 
-// TODO: replace this with a good example
-async function allCapsAsyncTest(s) {
+function hyphenate(s) {
+  return s.replace(/\s+/g, '-');
+}
+
+async function cardSearchBySet(s) {
+  const query = { 
+    q: 'set:' + s,
+    limit: 1,
+    sort: 'random'
+  };
+  return cardFinderSearch(query);    
+}
+
+async function cardSearchByText(s) {
+  const query = { 
+    q: 'text:' + s,
+    limit: 1,
+    sort: 'random'
+  };
+  return cardFinderSearch(query);  
+}
+
+async function cardSearchByType(s) {  
+  const query = { 
+    q: 't:' + s,
+    limit: 1,
+    sort: 'random'
+  };
+  return cardFinderSearch(query);
+}
+
+function getColorDescription(colorIdentity) {
+  const colorNames = {
+    W: 'white',
+    U: 'blue',
+    B: 'black',
+    R: 'red',
+    G: 'green',
+    WU: 'UW',
+    WR: 'RW',
+    WG: 'GW',
+    BR: 'RB',
+    UBG: 'BUG',
+    URG: 'RUG'
+  };
+
+  if (!colorIdentity) {
+    return 'colorless';
+  }
+
+  const colorDescription = colorIdentity.join('');
+  return colorNames[colorDescription] || colorDescription;
+}
+
+async function cardFinderSearch(query) {
+  const SEARCH_API_JSON_URL = 'https://goodgamery.com/api/mtg/card/json';
   return new Promise(resolve => {
-    console.log('waiting 1000ms...');
-    setTimeout(() => {
-      resolve(s.toUpperCase());
-    }, 1000);
-  });
+    request.get({ url: SEARCH_API_JSON_URL, qs: query }, (err, data, body) => {
+      if (err) {
+        console.warn('Error fetching card data: ' + err);        
+        resolve('NO RESULTS');
```

Comment:
I'd like to make this configurable so that when running the actual bot, this will fall back to a random card, and while developing, it will print an obvious message so that any issues with queries will be noticeable.

Reply:
You can check for environment variables in node - `process.env.NODE_ENV`. Will need to make sure in the deployed instance that the ENV is always `production`.